chain/chain/src/chain.rs

```diff
@@ -2143,6 +2146,8 @@ impl<'a> ChainUpdate<'a> {
                     let chunk =
                         self.chain_store_update.get_chunk_clone_from_header(&chunk_header)?;
 
+                    validate_transactions_order(&chunk.transactions)?;
```

Comment:
let's also roll in the condition under this line (the check_blocks_on_same_chain) and rename to `validate_chunk_transactions`.

and you need to use `near_chain::ErrorKind::InvalidChunkProofs` to return the chunk and related information for creation of challenge.
And then make sure in `test_verify_chunk_invalid_proofs_challenge` also call `validate_chunk_transactions`.

Reply:
Done